cns/restserver/ipam.go

```diff
@@ -127,7 +127,20 @@ func (service *HTTPRestService) requestIPConfigHandler(w http.ResponseWriter, r
 		logger.ResponseEx(service.Name+operationName, ipconfigsRequest, reserveResp, reserveResp.Response.ReturnCode, err)
 		return
 	}
-
+	// Checks to make sure we return exactly 1 IP
+	if len(ipConfigsResp.PodIPInfo) != 1 {
```

Comment:
not clear why this code is required

Reply:
This check is looking for the situation that ipam assigns multiple IPs due to multiple NCs but since we are using the old API we will only return 1. This could potentially happen if we have multiple NCs but we are using the old API. In that situation IPAM will still try to assign multiple IPs and then only the first will be returned from the list of IPs assigned back to the client. 

To prevent this I added an error check so we catch when IPAM is trying to assign multiple IPs when using the old API.

Reply:
@rjdenney Can you put basically exactly what you wrote as a comment above that conditional? If @ashvindeodhar has that question now, it's certain that a future reader of the code will as well.

Reply:
yeah I'll add it